go/tendermint/abci/mux.go

```diff
@@ -200,19 +194,29 @@ func (a *ApplicationServer) RegisterGenesisHook(hook func()) {
 	a.mux.registerGenesisHook(hook)
 }
 
+// RegisterHaltHook registers a function to be called when the
+// consensus Halt epoch height is reached.
+func (a *ApplicationServer) RegisterHaltHook(hook func(ctx context.Context, blockHeight int64, epoch epochtime.EpochTime)) {
+	a.mux.registerHaltHook(hook)
+}
+
 // Pruner returns the ABCI state pruner.
 func (a *ApplicationServer) Pruner() StatePruner {
 	return a.mux.state.statePruner
 }
 
+func (a *ApplicationServer) SetEpochtime(epochTime epochtime.Backend) {
+	a.mux.state.timeSource = epochTime
```

Comment:
Maybe add a comment somewhere that this must be set before the mux can be used? Also add a documentation comment for this method.